lib/ramble/ramble/test/application_language.py

```diff
@@ -188,13 +239,19 @@ def add_default_compiler(app_inst, spec_num=1):
         'compiler_spec': spec_compiler_spec
     }
 
-    default_compiler(spec_name, spack_spec=spec_spack_spec,  # noqa: F405
-                     compiler_spec=spec_compiler_spec)(app_inst)
+    if func_type == func_types.directive:
```

Comment:
Do you think it might make sense to parametrize this instead of repeating the logic from lines 224-231, or would that be overly complex or impact readability?

Reply:
I considered this when I implemented it, and decide it was not worth the trouble to parameterize it. The trouble specifically is that the `method` syntax is calling a `class method`, while the other `directive` syntax calls an arbitrary function, and the class is an input into it.